### PR TITLE
Refactored Bug fix. failed if it was user id in string format

### PR DESCRIPTION
**Pull request by joseffb-mla**
Bug Fix: ID failed if it was user id in string format i.e. '120029. Need to account for Hash/Gloabl ID's as well as numeric.

### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [ ] Make sure you are requesting to pull request from a **topic/feature/bugfix branch** (right side). Don't pull request from your master!

What does this implement/fix? Explain your changes.
---------------------------------------------------
…

Does this close any currently open issues?
------------------------------------------
…

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Operating System:** …

**WordPress Version:** …

### CODE REVIEW

File: src/Mutation/PostObjectCreate.php [hunk 1]
```diff
@@ -211,8 +212,16 @@ public static function mutate_and_get_payload( $post_type_object, $mutation_name
             /**
              * If the post being created is being assigned to another user that's not the current user, make sure
              * the current user has permission to edit others posts for this post_type
-             */
-            if ( ! empty( $input['authorId'] ) && get_current_user_id() !== $input['authorId'] && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
+             *
+	     *
+	     * Bug Fix: ID failed if it was user id in string format i.e. '120029. Need to account for Hash/Gloabl ID's as well as numeric.
+	     */
+	    $authorID = absint($input['authorId'])
```
Comment (jasonbahl) [inline src/Mutation/PostObjectCreate.php]: looks like there's a syntax error in here causing all tests to fail. 

I think the missing `;` on line 219 is an issue for sure

Then spacing on line 220 also needs adjusted

Comment (jasonbahl) [reply src/Mutation/PostObjectCreate.php]: @joseffb-mla (mentioning you to make sure you get notified)

**Comment by joseffb-mla** [comment]
Ah, maybe the inline github editor os t best after all due to copy paste.
Ill fix it up and resub monday.

File: src/Mutation/PostObjectCreate.php [hunk 2]
```diff
@@ -208,11 +209,16 @@ public static function mutate_and_get_payload( $post_type_object, $mutation_name
                 throw new UserError( sprintf( __( 'Sorry, you are not allowed to create %1$s', 'wp-graphql' ), $post_type_object->graphql_plural_name ) );
             }
 
-            /**
+           /**
              * If the post being created is being assigned to another user that's not the current user, make sure
              * the current user has permission to edit others posts for this post_type
              */
-            if ( ! empty( $input['authorId'] ) && get_current_user_id() !== $input['authorId'] && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
+            $authorID = absint($input['authorId']);
```
Comment (CodeProKid) [inline src/Mutation/PostObjectCreate.php]: spacing around `()` needed

File: src/Mutation/PostObjectCreate.php [hunk 3]
```diff
@@ -208,11 +209,16 @@ public static function mutate_and_get_payload( $post_type_object, $mutation_name
...
+            if ($id_components = Relay::fromGlobalId ( $input['authorId'] ) && is_array( $id_components ) && ! empty( $id_components['id'])) {
+	            $authorID = $id_components['id'];
```
Comment (CodeProKid) [inline src/Mutation/PostObjectCreate.php]: Need to use tabs instead of spaces all over here.

File: src/Mutation/PostObjectCreate.php [hunk 4]
```diff
@@ -208,11 +209,16 @@ public static function mutate_and_get_payload( $post_type_object, $mutation_name
...
             }
 
-            /**
+           /**
              * If the post being created is being assigned to another user that's not the current user, make sure
              * the current user has permission to edit others posts for this post_type
              */
-            if ( ! empty( $input['authorId'] ) && get_current_user_id() !== $input['authorId'] && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
+            $authorID = absint($input['authorId']);
+            if ($id_components = Relay::fromGlobalId ( $input['authorId'] ) && is_array( $id_components ) && ! empty( $id_components['id'])) {
```
Comment (CodeProKid) [inline src/Mutation/PostObjectCreate.php]: some inconsistent spacing inside `()` here.

File: src/Mutation/PostObjectCreate.php [hunk 5]
```diff
@@ -208,11 +209,16 @@ public static function mutate_and_get_payload( $post_type_object, $mutation_name
...
+	            $authorID = $id_components['id'];
+	        }
+
+            if ( ! empty( $authorID ) && get_current_user_id() !== $authorID && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
```
Comment (CodeProKid) [inline src/Mutation/PostObjectCreate.php]: Please break up each of these checks on their own line to make this more readable. Like:
```suggestion
            if ( ! empty( $authorID ) && 
                get_current_user_id() !== $authorID && 
                ! current_user_can( $post_type_object->cap->edit_others_posts ) 
            ) {
```

**Comment by joseffb-mla** [comment]
need to re add it from my actual IDE. I've actually already refactored a lot as you've mentioned.

**Comment by jasonbahl** [comment]
@CodeProKid is alive!

**Pull request closed without merging**